app/lib/features/activities/widgets/invitation_card.dart

```diff
@@ -22,13 +22,33 @@ class InvitationCard extends ConsumerWidget {
   });
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
+  ConsumerState<ConsumerStatefulWidget> createState() => _InvitationCardState();
+}
+
+class _InvitationCardState extends ConsumerState<InvitationCard> {
+  String? roomTitle;
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchTitle();
+  }
+
+  void _fetchTitle() async {
+    final title = await widget.invitation.room().displayName();
```

Comment:
Why to use stateful widget and setting title first on initialisation?

`invitation.room().displayName();` doesn't work on ListTile title where we are actually showing title?

Reply:
Agree with @kumarpalsinh25 on this. Why having whole widget ```ConsumerStateful``` if ```ref``` is only needed for invalidating in widget methods? Could be simplified with just ```Consumer``` builder?

Reply:
invitations are slightly different. this was the quickest way to make this work... as the invitation changing would reset the widget anyways...